### PR TITLE
Refactor PGM writing to be more explicit.

### PR DESCRIPTION
**Pull request by SirVer**

### CODE REVIEW

File: cartographer_ros/cartographer_ros/ros_map_writing_points_processor.cc [hunk 1]
```diff
@@ -24,37 +24,39 @@ namespace cartographer_ros {
 
 namespace {
 
-void WritePgm(const ::cartographer::io::Image& image, const double resolution,
+void WritePgm(const double resolution, ::cartographer::io::Image* image,
               ::cartographer::io::FileWriter* file_writer) {
+  ::cartographer::io::UniqueCairoSurfacePtr surface = image->GetCairoSurface();
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/ros_map_writing_points_processor.cc]: Revert this? Seemed better before.

File: cartographer_ros/cartographer_ros/ros_map_writing_points_processor.cc [hunk 2]
```diff
@@ -24,37 +24,39 @@ namespace cartographer_ros {
...
+  CHECK_EQ(cairo_image_surface_get_format(surface.get()),
+           ::cartographer::io::kCairoFormat);
+
+  const uint32_t* pixel_data =
+      reinterpret_cast<uint32_t*>(cairo_image_surface_get_data(surface.get()));
+
   // Flipping the image into ROS coordinate frame.
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/ros_map_writing_points_processor.cc]: This seems outdated?

**Comment by SirVer** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user SirVer. Merge queue now has a length of 3.

**Pull request merged**